dpnp/backend_mathematical.pyx

```diff
@@ -108,6 +110,54 @@ cpdef dparray dpnp_copysign(dparray x1, dparray x2):
     return call_fptr_2in_1out(DPNP_FN_COPYSIGN, x1, x2, x1.shape)
 
 
+cpdef dparray dpnp_cumprod(dparray x1):
+
+    x1_dtype = x1.dtype
```

Comment:
I'm not sure variable `x1_dtype` is needed. `x1.dtype` may be used directly. The same is in `dpnp_cumsum`.